### PR TITLE
Adding flesh raptors to mainline

### PR DESCRIPTION
**Pull request by thelonestander**
I always thought Flesh raptors from DDA are cool so I added them to BN. 

Co-Authored-By: I-am-Erk <45136638+I-am-Erk@users.noreply.github.com>

#### Purpose of change

As I said ive loved the flesh raptors in DDA and since we were able to port this fix https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1865 I figured it would be nice to have a monster in game that uses this. 

#### Describe the solution

Just moved the files over from my mod and fixed up the spell shapes. 

#### Describe alternatives you've considered

Leaving it in my mod

#### Testing

Tested in a new game with new mods, spawned some fleshy boys in they work! 

#### Additional context

N/a

### CODE REVIEW

File: data/json/monstergroups/zombie_upgrades.json [hunk 1]
```diff
@@ -10,5 +10,46 @@
       { "monster": "mon_zombie_soldier_acid_2", "freq": 15, "cost_multiplier": 2 },
       { "monster": "mon_zombie_kevlar_1", "freq": 25, "cost_multiplier": 2 }
     ]
+  },
+  {
+    "type": "monstergroup",
+    "name": "GROUP_ZOMBIE_UPGRADE",
```
Comment (Coolthulhu) [inline data/json/monstergroups/zombie_upgrades.json:16]: Why are those groups "redefined" like this? It should probably be added to the original group. Same for the rest, below.

**Comment by LyleSY** [comment]
Great idea, just needs the tweaks Coolthulhu asked for and it should be fine to merge. As is, defining the same monster groups twice in the same file in mainline won't work like it does with a mod

**Comment by thelonestander** [comment]
Should be good to go now.

**Comment by Coolthulhu** [comment]
Tested them and found a bunch of issues:
* With 3D vision, the raptors can stay above player and won't go down to attack
* Pupating zombies can cause massive stunlocks in melee - I assume it's because of their sludge trail, which seems to apply move drain when it creates sludge under the player
* Raptors have evolutions, but pupating zombies do not - evolved raptors will only happen if the player gets near to spawn the raptors, then goes away to let them evolve
* Shady pupating zombies don't seem to have any spawn group or other non-debug way of existing
* (Possibly) Shadow raptors have that weird darkness invisibility - not sure if we want that

I don't think they are ready for BN in current state.

**Comment by Zlorthishen** [comment]
> With 3D vision, the raptors can stay above player and won't go down to attack
is that a json issue or a c++ issue?

**Comment by Coolthulhu** [comment]
C++, it needs some fixes to 3D monsters.

**Pull request closed without merging**